### PR TITLE
Rebase to 1.20 (Istio 1.12)

### PR DESCRIPTION
**Pull request by twghu**
This is the basis for Maistra 2.2 Envoy, merged from Upstream Envoy 1.20 onto Maistra main with pickings from Maistra 2.1.

### CODE REVIEW

**Comment by twghu** [comment]
It is intended this will be the new maistra-2.2 branch not necessarily merged into main.

Working on building with latest quay image.

Resolving inconsistencies in the tools folder.

**Comment by twghu** [comment]
Added 

`  --deleted_packages=test/common/quic,test/common/quic/platform \
  --//bazel:http3=false \
`
to the maistra/run-ci.sh

File: maistra/run-ci.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export BUILD_SCM_REVISION="Maistra PR #${PULL_NUMBER:-undefined}"
+export BUILD_SCM_STATUS="SHA=${PULL_PULL_SHA:-undefined}"
+
+# Build
+time bazel build \
+  --incompatible_linkopts_to_linklibs \
+  --local_ram_resources=12288 \
+  --local_cpu_resources=8 \
+  --jobs=4 \
```
Comment (jwendell) [inline maistra/run-ci.sh]: I recommend decreasing these to 6 and 3. Prow might kill the jobs. You'll see errors like
```
gcc: fatal error: Killed signal terminated program cc1plus
```

File: maistra/patches/modify-include-paths.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Comment (jwendell) [inline maistra/patches/modify-include-paths.sh]: I don't think this is necessary anymore

File: maistra/run-ci.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  //source/exe:envoy-static
+
+echo "Build succeeded. Binary generated:"
+bazel-bin/source/exe/envoy-static --version
+
+# Run tests
+time bazel test \
+  --incompatible_linkopts_to_linklibs \
+  --local_ram_resources=12288 \
+  --local_cpu_resources=8 \
```
Comment (jwendell) [inline maistra/run-ci.sh]: ditto

File: .bazelrc [hunk 4]
```diff
@@ -392,3 +392,14 @@ build:windows --dynamic_mode=off
...
 try-import %workspace%/local_tsan.bazelrc
+
+build --cxxopt -D_GLIBCXX_USE_CXX11_ABI=1
+build --cxxopt -DENVOY_IGNORE_GLIBCXX_USE_CXX11_ABI_ERROR=1
+build --cxxopt -Wnon-virtual-dtor
+build --cxxopt -Wformat
+build --cxxopt -Wformat-security
+build --cxxopt -Wno-error=deprecated-declarations
+build --cxxopt -Wno-error=unused-variable
+build --cxxopt -w
```
Comment (jwendell) [inline .bazelrc:403]: I had issues with this in proxy. It was hiding some warnings and messing with some packages that rely on warnings to configure the build. So I removed this flag from proxy.

File: bazel/foreign_cc/BUILD [hunk 5]
```diff
@@ -39,32 +39,6 @@ cc_library(
     ],
 )
 
-# Kafka client dependency used by Kafka-mesh filter.
```
Comment (jwendell) [inline bazel/foreign_cc/BUILD]: Why this was removed?

File: bazel/repository_locations.bzl [hunk 6]
```diff
@@ -1056,6 +1035,29 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         sha256 = "ce7a5c76e8b425aca874cea329fd9ac44b203b777053453b6a37b4496c5ce34f",
         strip_prefix = "emsdk-{version}",
         urls = ["https://github.com/emscripten-core/emsdk/archive/{version}.tar.gz"],
+        patch_cmds = [
```
Comment (jwendell) [inline bazel/repository_locations.bzl:1039]: nit: Add something like
```
FIXME: https://issues.redhat.com/browse/OSSM-781
```

Ideally we would need to verify if 2.0.7 is still the version used by envoy 1.20. This was true for 1.17.

File: maistra/run-ci.sh (hunk 3 above)
Comment (twghu) [reply maistra/run-ci.sh]: fixed

File: maistra/run-ci.sh (hunk 1 above)
Comment (twghu) [reply maistra/run-ci.sh]: fixed

File: bazel/repository_locations.bzl (hunk 6 above)
Comment (twghu) [reply bazel/repository_locations.bzl:1039]: Commented, and will follow up on the issue.

**Review by jwendell — APPROVED** [review]

**Review by oschaaf — APPROVED** [review]

**Pull request merged**